setup.py

```diff
@@ -28,13 +31,15 @@
 try:
     import ninja # noqa
     WITH_NINJA = True
+    WITH_NINJA = False
```

Comment:
we can remove all these NINJA stuffs, basically delegate the ninja generation work to cmake and we only check whether `ninja` command is available.